### PR TITLE
 GJets_1j_Gpt_5f_NLO ->  5 bins between 30 to Inf

### PR DESCRIPTION
**Pull request by hbecerri**

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-150To250_5f_NLO/GJets_1j_Gpt-150To250_5f_NLO_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
+
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-150To250_5f_NLO/GJets_1j_Gpt-150To250_5f_NLO_proc_card.dat:6]: please remove those lines, the p/j definition already include b quarks if *_no_b_mass model is imported

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-250To400_5f_NLO/GJets_1j_Gpt-250To400_5f_NLO_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
+
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-250To400_5f_NLO/GJets_1j_Gpt-250To400_5f_NLO_proc_card.dat:6]: please remove those lines, the p/j definition already include b quarks if *_no_b_mass model is imported

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-30To150_5f_NLO/GJets_1j_Gpt-30To150_5f_NLO_proc_card.dat [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
+
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-30To150_5f_NLO/GJets_1j_Gpt-30To150_5f_NLO_proc_card.dat:6]: please remove those lines, the p/j definition already include b quarks if *_no_b_mass model is imported

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-400To650_5f_NLO/GJets_1j_Gpt-400To650_5f_NLO_proc_card.dat [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
+
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-400To650_5f_NLO/GJets_1j_Gpt-400To650_5f_NLO_proc_card.dat:6]: please remove those lines, the p/j definition already include b quarks if *_no_b_mass model is imported

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-650ToInf_5f_NLO/GJets_1j_Gpt-650ToInf_5f_NLO_proc_card.dat [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
+
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/GJets_1j_Gpt-650ToInf_5f_NLO/GJets_1j_Gpt-650ToInf_5f_NLO_proc_card.dat:6]: please remove those lines, the p/j definition already include b quarks if *_no_b_mass model is imported

**Review by menglu21 — COMMENTED** [review]
besides the comments listed in each proc_card, could you please also perform a check between the inclusive Gpt (from 30 - inf) and the one by adding all those separate pt range, e.g., the gamma pt distribution

**Comment by hbecerri** [comment]
Dear expertes,

Sorry for the delay. As suggested,  we made the plots showing the pT distribution of the photon for the inclusive GpT (30-Inf) sample and the combination of the different pT range samples (see the attachment). We don't see the problem mentioned in the presentation in the current MG version. Do you have any further comments?

Thanks, 
Ying and Hugo

selection_0
selection_0

**Comment by menglu21** [comment]
Hi @hbecerri thanks for the check and plots, i had a discussion with @AnYpku about those cards, since there are already cards https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/GJets_Gpt_1j_5f_NLO, which seems to be the same with yours, i suggest to update [make_cards.py](https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/GJets_Gpt_1j_5f_NLO/make_cards.py) to make it include the pt-range you need.

N.B., there are already G1jet samples, which has e.g., 130-270 in the run cards for gridpack generation, and LHEFilter 150-250 in the sample fragment to implement the actual pt range they want. so when updating the make_cards.py, please make the range consistent with the values used in the gridpack generation.

**Comment by hbecerri** [comment]
Hello @menglu21. Thanks for the example. I have updated the cards to extend the range of the bins leaving an space of ~30TeV per bin. The new pT bins are: 
20-130
70-230
170-430
370-700
650-Inf 

So the idea is to apply the LHEFilters to have 
30-100
100-200
200-400
400-675
675-Inf

**Comment by menglu21** [comment]
Hi @hbecerri thanks for the update. one thing is that you are missing "lhapdf    = pdlabel   ! PDF set " in therun cards. One last comment is could you please make the PR to repository "cards/production/2017/13TeV", maybe with name "GJets_1j_new_Gpt-bin_5f_NLO", and put all your cards inside (to make them different from the existing "GJets_1j_Gpt-100To250_5f_NLO")

**Comment by menglu21** [comment]
@hbecerri thanks, cards look good to me. @Saptaparna @mlizzo please have a look

**Pull request merged**